### PR TITLE
add a script for easy cluster setup

### PR DESCRIPTION
**Pull request by KeitaW**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by perifaws** [comment]
LGTM

**Review by perifaws — APPROVED** [review]

File: 1.architectures/5.sagemaker-hyperpod/easy-setup.sh [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+            ;;
+        esac
+    done
+}
+
+parse_args $@
+
+mkdir $CLUSTER_NAME && cd $CLUSTER_NAME
+# Check for JQ
+if ! command -v jq &> /dev/null
```
Comment (sean-smith) [inline 1.architectures/5.sagemaker-hyperpod/easy-setup.sh:87]: check for aws cli as well.

**Comment by KeitaW** [comment]
Thanks @perifaws  for reviewing and @sean-smith  for the comments. I'll address the comments today.

File: 1.architectures/5.sagemaker-hyperpod/easy-setup.sh [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    ],
+    "VpcConfig": {
+      "SecurityGroupIds": ["$SECURITY_GROUP"],
+      "Subnets":["$SUBNET_ID"]
+    }
+}
+EOL
+
+echo "Now that you have the necessary files in the S3 bucket, you can create the cluster by running the following command:"
+echo "aws sagemaker create-cluster --cli-input-json file://cluster-config.json" ${REGION}
```
[the remark this answers is not in the document]
Comment (KeitaW) [reply 1.architectures/5.sagemaker-hyperpod/easy-setup.sh]: Sure https://github.com/aws-samples/awsome-distributed-training/pull/282/commits/b87d6cef5f1bc622eebb30f26f7bdcbacf0931cf

File: 1.architectures/5.sagemaker-hyperpod/easy-setup.sh [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        "ExecutionRole": "${ROLE}",
+        "ThreadsPerCore": 1
+      }
+    ],
+    "VpcConfig": {
+      "SecurityGroupIds": ["$SECURITY_GROUP"],
+      "Subnets":["$SUBNET_ID"]
+    }
+}
+EOL
```
[the remark this answers is not in the document]
Comment (KeitaW) [reply 1.architectures/5.sagemaker-hyperpod/easy-setup.sh:327]: addressed in https://github.com/aws-samples/awsome-distributed-training/pull/282/commits/7130031f7c27adadf016356cc6672579ff5e2ddb

File: 1.architectures/5.sagemaker-hyperpod/easy-setup.sh (hunk 1 above)
Comment (KeitaW) [reply 1.architectures/5.sagemaker-hyperpod/easy-setup.sh:96]: Check added in https://github.com/aws-samples/awsome-distributed-training/pull/282/commits/ed3c0af46734979edc47336a6dc172cdbd590908

**Review by sean-smith — APPROVED** [review]
LGTM, let's incorporate this into the workshop somehow. 

I want to keep some flexibility around customizations to cluster, i.e. adding additional compute queues but this seems to be a good "easy path" for most customers. 

Nice work.

**Comment by KeitaW** [comment]
Thanks Sean. I merge this and create another PR on the customization.

**Pull request merged**